app/lib/tutorial.cjsx

```diff
@@ -111,3 +105,29 @@ module.exports = React.createClass
 
   handleNextClick: ->
     @refs.stepThrough.goNext()
+
+  handleUnmount: ->
+    now = new Date().toISOString()
+    completedThisSession[@props.tutorial.id] = now
+
+    if @props.user?
+      projectPreferences = @props.preferences
+      projectPreferences ?= apiClient.type('project_preferences').create
+        links:
+          project: project.id
+        preferences: {}
+      # Build this manually. Having an index (even as a strings) keys creates an array.
+      projectPreferences.preferences ?= {}
+      projectPreferences.preferences.tutorials_completed_at ?= {}
+      projectPreferences.update "preferences.tutorials_completed_at.#{@props.tutorial.id}": now
+      projectPreferences.save()
+
+      @logToGeordi now
```

Comment:
Currently, this will only hit `logToGeordi` if the user is logged in, due to the conditional above. Is this intentional that we only want to log if a user is present?